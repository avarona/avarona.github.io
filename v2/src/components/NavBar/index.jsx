import React from "react";
import styles from "./styles.module.scss";
import headshot from "../../images/headshot.png";
import { Icon } from "@avarona/react-components";

const NavBar = () => (
  <nav className={styles.navContainer}>
    <Icon addClass={styles.image} img={headshot} alt='headshot' size={40} />

    <ul className={styles.navList}>
      <li className={styles.navItem}>
        <a href='#portfolio'>Projects</a>
      </li>
      <li className={styles.navItem}>
        <a href='#about'>About</a>
      </li>
      <li className={styles.navItem}>
        <a href='#github'>Github</a>
      </li>
      <li className={styles.navItem}>
        <a href='#linkedin'>LinkedIn</a>
      </li>
      <li className={styles.navItem}>
        <a href='#angel'>Angel.co</a>
      </li>
      <li className={styles.navItem}>
        <a href='#contact'>Contact</a>
      </li>
    </ul>
  </nav>
);

export default NavBar;
