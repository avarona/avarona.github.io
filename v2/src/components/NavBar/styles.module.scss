.navContainer {
  position: fixed;
  width: 100%;
  box-shadow: 0 1px 1px #e5eff5;
  font-size: 18px;
}

.wrapper {
  display: flex;
  align-items: center;
  padding: 0 48px;
}

.image {
  width: 40px;
  height: 40px;
  border-radius: 50%;
}

.navList {
  display: flex;
  margin: 0 0 0 auto;
}

.navItem {
  padding: 0 8px;
  list-style-type: none;
  align-self: flex-end;
  padding: 24px 8px;
  transition: background 0.2s ease-in-out;
  cursor: pointer;

  a {
    text-decoration: none;
    color: red;
  }

  &:hover {
    background: white;
  }
}
