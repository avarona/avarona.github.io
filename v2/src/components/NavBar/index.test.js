import React from "react";
import { shallow } from "enzyme";
import NavBar from "./";

describe("NavBar Component", () => {
  it("renders", () => {
    const wrapper = shallow(<NavBar />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
