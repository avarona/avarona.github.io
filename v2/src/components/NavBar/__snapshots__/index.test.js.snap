// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component renders 1`] = `"<nav class=\\"navContainer\\"><div class=\\"wrapper\\"><img class=\\"image styles-module_icon__8puW5\\" src=\\"headshot.png\\" alt=\\"headshot\\"/><ul class=\\"navList\\"><li class=\\"navItem\\"><a href=\\"#portfolio\\">Projects</a></li><li class=\\"navItem\\"><a href=\\"#about\\">About</a></li><li class=\\"navItem\\"><a href=\\"#github\\">Github</a></li><li class=\\"navItem\\"><a href=\\"#linkedin\\">LinkedIn</a></li><li class=\\"navItem\\"><a href=\\"#angel\\">Angel.co</a></li><li class=\\"navItem\\"><a href=\\"#contact\\">Contact</a></li></ul></div></nav>"`;
